modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM614AddDocumentationToAnAPIWithDocTypeSampleAndSDKThroughPublisherRestAPITestCase.java

```diff
@@ -140,10 +104,20 @@ public void testApiCreation() throws Exception {
         apiCreationRequestBean.setTechOwner("api620t");
         apiCreationRequestBean.setTechOwnerMail("api620t@ww.com");
 
-        JSONObject jsonObject = new JSONObject(apiPublisher.
-                addAPI(apiCreationRequestBean).getData());
-        assertFalse(jsonObject.getBoolean("error"), apiName + " is not created ");
+        APIDTO apiDto = createAndPublishAPI(apiCreationRequestBean, restAPIPublisher, false);
+        apiId = apiDto.getId();
 
+        documentDTO.setName(docName);
+        documentDTO.setSummary(summary);
+        documentDTO.setType(DocumentDTO.TypeEnum.HOWTO);
+        documentDTO.setSourceType(DocumentDTO.SourceTypeEnum.FILE);
+        documentDTO.setVisibility(DocumentDTO.VisibilityEnum.API_LEVEL);
+
+        org.wso2.carbon.automation.test.utils.http.client.HttpResponse documentationResponse =
+                restAPIPublisher.addDocument(apiId, documentDTO);
+        assertEquals(documentationResponse.getResponseCode(), 200,
+                "Error while add documentation to API");
```

Comment:
AFAIU, this particular test case is to verify file-based doc upload. Hence, isn't it better to indicate this in the assertion error as well?

Reply:
Fixed with https://github.com/wso2/product-apim/pull/8604